src/frontends/pytorch/src/op/scaled_dot_product_attention.cpp

```diff
@@ -99,11 +101,43 @@ OutputVector translate_scaled_dot_product_attention(const NodeContext& context)
         }
         scaled_atten = context.mark_node(std::make_shared<v1::Add>(scaled_atten, atten_mask));
     }
-    scaled_atten = context.mark_node(std::make_shared<v8::Softmax>(scaled_atten, -1));
-    return {context.mark_node(std::make_shared<v0::MatMul>(scaled_atten, value))};
+    if (!(scaled_atten->is_dynamic()) && scaled_atten->get_output_partial_shape(0).rank() == 4) {
+        const std::vector<size_t>* scaled_atten_v = &(scaled_atten->get_shape());
+        std::vector<size_t> softmax_shape_v((*scaled_atten_v).begin() + 1, (*scaled_atten_v).end());
+        softmax_shape_v[0] *= (*scaled_atten_v)[0];
+        std::vector<size_t> matmul_shape_v(*scaled_atten_v);
+        auto softmax_shape = context.mark_node(v0::Constant::create(element::i32, Shape{3}, softmax_shape_v));
+        auto reshape_3d = context.mark_node(std::make_shared<v1::Reshape>(scaled_atten, softmax_shape, false));
+        scaled_atten = context.mark_node(std::make_shared<v8::Softmax>(reshape_3d, -1));
+        auto matmul_shape = context.mark_node(v0::Constant::create(element::i64, Shape{4}, matmul_shape_v));
+        scaled_atten = context.mark_node(std::make_shared<v1::Reshape>(scaled_atten, matmul_shape, false));
```

Comment:
```suggestion
    auto satten_out_pshape = scaled_atten->get_output_partial_shape(0);
    if (satten_out_pshape.is_static() && satten_out_pshape.rank().get_length() == 4) {
        auto scaled_atten_v = satten_out_pshape.to_shape();
        Shape softmax_shape_v(scaled_atten_v.begin() + 1, scaled_atten_v.end());
        softmax_shape_v[0] *= scaled_atten_v[0];
        auto softmax_shape = context.mark_node(v0::Constant::create(element::i32, Shape{3}, softmax_shape_v));
        auto reshape_3d = context.mark_node(std::make_shared<v1::Reshape>(scaled_atten, softmax_shape, false));
        scaled_atten = context.mark_node(std::make_shared<v8::Softmax>(reshape_3d, -1));
        auto matmul_shape = context.mark_node(v0::Constant::create(element::i32, Shape{4}, scaled_atten_v));
        scaled_atten = context.mark_node(std::make_shared<v1::Reshape>(scaled_atten, matmul_shape, false));
```
That should work, but I didn't check

Reply:
Seems to be working as expected